### DISCUSSION TITLE
Has anybody had any luck setting up push-to-talk in Discord?

### DISCUSSION THREAD

**Discussion by Hubro** [hyprwm/Hyprland#2083] [Q&A] [answered]
I have set up ALT as my Discord push-to-talk key. This works while Discord is focused, as well as any other X11 window. It doesn't work when having a Wayland window focused, however. Reading [the docs for the `pass` dispatcher](https://wiki.hyprland.org/Configuring/Dispatchers/), I was hopeful this might be a solution.

I've tried configuring it like this:

```
# Push-to-talk in Discord?
bind = ALT, , pass, ^discord$
```

Which doesn't crash, but also nothing happens. Any idea if it's possible to get this to work?

**Comment by vaxerski** [comment, marked as answer]
`ALT, Alt_L`

ref https://wiki.hyprland.org/Configuring/Binds/#binding-mods

**Reply by Hubro** [reply]
Ach sorry, I missed that. It still doesn't seem to work though. Should binding mods alone still work with `pass`, or is that perhaps a limitation?

EDIT: Yeah it seems to be a limitation with binding mods, if I use a different key like <kbd>F6</kbd> then it works.

**Reply by vaxerski** [reply]
Might be. You'll probably have to wait for me to implement transparent binds or something, possibly. IIRC modbinds are only available with bindr, which would not work with something like ptt.

**Reply by Hubro** [reply]
After a bit more testing, it seems global keybinds (`pass`) only works with single keys. Binding something like:

```
bind = ALT, less, pass, ^discord$
```

Also does not work. It seems like either ALT or less gets pressed in the Discord window, but never released. When I re-enter the Discord window, I have to press both buttons once to "unstick" them :smile: 

Transparent binds sounds awesome. After having used Sway for years now, having global keybinds sounds like a magical prospect, a huge leap ahead for Wayland :grin: I really hope you figure it out!

**Reply by vaxerski** [reply]
transparent just means unshadowable.

mod-binds not working might be an electron issue, in other apps it works, but I'll take a look at it.

**Reply by Buttars** [reply]
Any update on this? Discord still consumes my keys and never released them.

`bindn = ,c, pass, ^(discord)$`

Anytime `c` is pressed anywhere except discord it sticks on and only is released when `c` is pressed with the discord window in focus.

Weirdly enough the `c` key gets repeated in the chat. When I press backspace it stops repeating `c` but the ptt still shows active like the key is still being pressed.

It's almost like the mapping is canceled with backspace for some reason but discord doesn't pick up the key event probably because it wasn't the `c` key that terminated the mapping?

**Reply by muni-corn** [reply]
Do you need to use `bindn`? I've binded mine to `Print` with just `bind` and it works fine

**Reply by Buttars** [reply]
I need a global keybind that is passed to the focused application and Discord. I believe bindn is the appropriate option for that.
